examples/multi-threading.py

```diff
@@ -17,26 +22,29 @@ def onLoggedIn(self, email=None):
         print("Login of {} successful.".format(email))
 
 
-# Logging in and setting logging level to WARNING to avoid some unessential output
-client = PrintMessage("<email>", "<password>", logging_level=30)
+# Login and set logging level to WARNING to avoid some unessential output
+client1 = Client(user, password, logging_level=logging.WARNING)
+client2 = MessagePrinter(user, password, logging_level=logging.WARNING)
+
+
+# Creating and starting a separate thread for receiving messages
+t1 = threading.Thread(target=client2.listen, daemon=True)
+t1.start()
 
 
-def send():
+# Loop checking for, and sending messages
+try:
     while True:
         payload = input("Message: ")
         if payload:
-            client.send(
-                Message(text=payload), thread_id=client.uid, thread_type=ThreadType.USER
+            client1.send(
+                Message(text=payload),
+                thread_id=client1.uid,
+                thread_type=ThreadType.USER,
             )
 
-
-def receive():
-    while True:
-        client.doOneListen()
-
-
-# Creating and starting separate threads for handling the receiving and sending of messages
-t1 = threading.Thread(target=receive)
-t2 = threading.Thread(target=send)
-t1.start()
-t2.start()
+# Clean-up on exit
+except KeyboardInterrupt:
+    client1.logout()
+    client2.logout()
+    sys.exit(0)
```

Comment:
We should clean the `client2` up properly instead. That can be done by setting `Client.listening` to `False`, and then wait for the thread to clean up afterwards, using `Thread.join`

Reply:
And then there'd be no need to make `t1` daemonic